File: miracle-domain/src/main/java/com/depromeet/domain/record/repository/RecordRepositoryCustom.java
package com.depromeet.domain.record.repository;

import com.depromeet.domain.record.Record;

import java.time.LocalDateTime;
import java.util.List;

public interface RecordRepositoryCustom {

    Record findByMemberIdAndScheduleIdAndStartTime(Long memberId, Long scheduleId, LocalDateTime startDateTime, LocalDateTime endDateTime);

    List<Record> findRecordBetween(Long memberId, LocalDateTime startDateTime, LocalDateTime endDateTime);

}

File: miracle-domain/src/main/java/com/depromeet/domain/schedule/record/RecordRepositoryCustom.kt
package com.depromeet.domain.schedule.record

import org.springframework.data.jpa.repository.Query


interface RecordRepositoryCustom {


    @Query(
        value = "SELECT * FROM record  WHERE member_id = ?1 and schedule_id = ?2 and year = ?3 and month = ?4 and day = ?5 ",
        nativeQuery = true
    )
    fun findByMemberIdAndScheduleIdAndYearAndMonthAndDay(
        memberId: Long,
        scheduleId: Long,
        year: Int,
        month: Int,
        day: Int
    ): MutableList<Record>?

}
